### ISSUE TITLE
Set tls-san option to the array of control-plane node IPs both public and private

### ISSUE THREAD

**Issue by mysticaltech** [kube-hetzner/terraform-hcloud-kube-hetzner#325]
### Discussed in https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/324

<div type='discussions-op-text'>

<sup>Originally posted by **captnCC** September 27, 2022</sup>
Hello folks out there,
I'm currently playing around with this module to create clusters and deploy applications on it.

The straight forward path seemed to be to use the cluster module together with the Kubernetes provider.

So I take the retuned `kubeconfig` from the cluster module and put it into the Kubernetes provider:
```terraform
provider "kubernetes" {
  host                   = module.cluster.kubeconfig.host
  client_certificate     = module.cluster.kubeconfig.client_certificate
  client_key             = module.cluster.kubeconfig.client_key
  cluster_ca_certificate = module.cluster.kubeconfig.cluster_ca_certificate
}
```

This works well after the cluster is created, but on the first run most resources will fail with an cert error  
`Failed to create deployment: Post "https://142.132.239.194:6443/apis/apps/v1/namespaces/default/deployments": x509: certificate is valid for 10.255.0.101, 10.43.0.1, 127.0.0.1, ::1, not 142.132.239.194`. 
Weirdly enough in some instances a part of the resources would succeed while the rest fails. Even adding a delay of 3 minutes before creating the resources didn't resolve that issue.

As base of the infrastructure I needed some additional components to be installed via helm. So I take the same config and put it in the helm provider:
```terraform
provider "helm" {
  kubernetes {
    host                   = module.cluster.kubeconfig.host
    client_certificate     = module.cluster.kubeconfig.client_certificate
    client_key             = module.cluster.kubeconfig.client_key
    cluster_ca_certificate = module.cluster.kubeconfig.cluster_ca_certificate
  }
}
```

And to my surprise this just works fine. 

Has anyone faced the same issue and what is your workaround? And the more general question: Keep infra and application in the same place or split it up in 2 separated terraform workflows? 

#### Update

By playing around with the order and dependencies of resources I found that generally the first request(s) to the Kubernetes API fail both with Kubernetes or Helm resources. </div>

**Comment by captnCC** [comment]
Thanks @mysticaltech for the hint, this was an easy fix, but making the setup much more reliable!

**Comment by mysticaltech** [comment]
Released in v1.5.5 🚀
